src/Bindings/PluginLua.cpp

```diff
@@ -72,12 +72,13 @@ void cPluginLua::Close(void)
 	cResettablePtrs resettables;
 	std::swap(m_Resettables, resettables);
 	{
-		cCSUnlock Unlock(Lock);
+		m_CriticalSection.Unlock();
 		for (auto resettable: resettables)
 		{
 			resettable->Reset();
 		}
 		m_Resettables.clear();
+		m_CriticalSection.Lock();
```

Comment:
Then either we can't have the annotations, or we need to use noexcept and try catch blocks as an alternatives. The annotations don't understand aliasing, so they can't tell that the inner reference is the same as the one passed in.